log/handlers_test.ts

```diff
@@ -0,0 +1,56 @@
+import { assertEqual, test } from "../testing/mod.ts";
+import { LogRecord, Logger } from "./logger.ts";
+import { LogLevel } from "./levels.ts";
+import { BaseHandler } from "./handlers.ts";
+
+class TestHandler extends BaseHandler {
+  public messages: string[] = [];
+
+  public log(str: string): void {
+    this.messages.push(str);
+  }
+}
+
+class CustomHandler extends TestHandler {
+  public handle(record: LogRecord): void {
+    return this.log(`testing ${record.msg} [${record.args.join(",")}]`);
+  }
+}
+
+test(function simpleHandler() {
```

Comment:
Test checking that records with lower level than handler's is not outputted would be great

Reply:
This test does that, it is asserting that the resulting messages are only those we asked for (see how `handler.messages` is checked further down and _does not_ contain `"DEBUG bar"`).

Reply:
Yep, that's right, but it'd be good to set against all possible combinations